### PR TITLE
Split out arrow-json (#3044)

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Part of #3044

# Rationale for this change
 

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: arrow/Cargo.toml [hunk 1]
```diff
@@ -64,9 +63,8 @@ packed_simd = { version = "0.3", default-features = false, optional = true, pack
 chrono = { version = "0.4", default-features = false, features = ["clock"] }
 comfy-table = { version = "6.0", optional = true, default-features = false }
 pyo3 = { version = "0.17", default-features = false, optional = true }
-lexical-core = { version = "^0.8", default-features = false, features = ["write-integers", "write-floats", "parse-integers", "parse-floats"] }
 multiversion = { version = "0.6.1", default-features = false }
-bitflags = { version = "1.2.1", default-features = false }
```
Comment (tustvold) [inline arrow/Cargo.toml:69]: These are just drive-by-fixes. They are no longer needed by the top-level crate

File: arrow-json/src/reader.rs [hunk 2]
```diff
@@ -23,12 +23,11 @@
 //!
 //! Example:
 //!
-//! ```
-//! use arrow::datatypes::{DataType, Field, Schema};
-//! use arrow::json;
-//! use std::fs::File;
-//! use std::io::BufReader;
-//! use std::sync::Arc;
+//! ```no_run
```
Comment (viirya) [inline arrow-json/src/reader.rs:26]: Wondering why change to `no_run`?

File: arrow-json/src/reader.rs [hunk 3]
```diff
@@ -2076,12 +2078,8 @@ mod tests {
 
     #[test]
     fn test_invalid_json_infer_schema() {
-        let re = infer_json_schema_from_seekable(
-            &mut BufReader::new(
-                File::open("test/data/uk_cities_with_headers.csv").unwrap(),
-            ),
-            None,
-        );
```
Comment (viirya) [inline arrow-json/src/reader.rs:2084]: We drop this file?

File: arrow-json/src/reader.rs (hunk 2 above)
Comment (tustvold) [reply arrow-json/src/reader.rs:26]: Oops need to revert

File: arrow-json/src/reader.rs (hunk 3 above)
Comment (tustvold) [reply arrow-json/src/reader.rs:2084]: No it just isn't part of the arrow-json test data

**Review by viirya — APPROVED** [review]
